paddle/phi/kernels/gpu/nanmedian_grad_kernel.cu

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/phi/kernels/nanmedian_grad_kernel.h"
+
+#include "paddle/fluid/platform/device/gpu/gpu_launch_config.h"
+#include "paddle/fluid/platform/device/gpu/gpu_primitives.h"
+#include "paddle/phi/backends/gpu/gpu_context.h"
+#include "paddle/phi/core/kernel_registry.h"
+#include "paddle/phi/core/tensor_meta.h"
+
+namespace phi {
+
+using paddle::platform::PADDLE_CUDA_NUM_THREADS;
+inline int GET_BLOCKS(const int N) {
+  return (N + PADDLE_CUDA_NUM_THREADS - 1) / PADDLE_CUDA_NUM_THREADS;
+}
+
+template <typename T>
+__global__ void KernelNanmedianGrad(const T* x_ptr,
+                                    const T* medians_ptr,
+                                    const T* out_grad_ptr,
+                                    T* x_grad_ptr,
+                                    int64_t stride,
+                                    int64_t numel) {
+  auto zero = static_cast<T>(0);
+  CUDA_KERNEL_LOOP(index, numel) {
+    int64_t row = static_cast<int64_t>(index / stride);
+    int64_t m_row = 2 * row;
+    if (isnan(static_cast<float>(x_ptr[index])) ||
+        isnan(static_cast<float>(medians_ptr[m_row])) ||
+        (x_ptr[index] != medians_ptr[m_row] &&
+         x_ptr[index] != medians_ptr[m_row + 1])) {
+      x_grad_ptr[index] = zero;
+    } else {
+      x_grad_ptr[index] = out_grad_ptr[row];
```

Comment:
this logic to get the gradient of nanmedian is not correct, discribed above

Reply:
I will use the index of sorted elements, and determine which one should compute the gradient. A question here, before I designed this kernel , the sorting algorithm did not work in case of nan values. Since last sharing of nanquantile in PFCC,  according to my knowledge,  this sort api is already fixed by someone. So that means, I can feel free to use sort api with no need to replace nan value with minimum. am I correct?

Reply:
done

Reply:
> Since last sharing of nanquantile in PFCC, according to my knowledge, this sort api is already fixed by someone. So that means, I can feel free to use sort api with no need to replace nan value with minimum. am I correct?
yes, sort api is updated recently, if have any use problem, you can issue or even start another pr to fix it:)